shepherd/storage/minio_storage.py

```diff
@@ -0,0 +1,52 @@
+from io import BytesIO
+
+from minio import Minio
+
+from shepherd.constants import DONE_FILE, ERROR_FILE
+from shepherd.storage.storage import Storage
+from shepherd.utils import minio_object_exists, pull_minio_bucket, push_minio_bucket
+
+
+class MinioStorage(Storage):
+    """
+    A remote storage adapter that uses the S3 client from Minio SDK.
+    """
+
+    def __init__(self, minio: Minio):
+        self._minio = minio
+
+    def is_accessible(self) -> bool:
+        try:
+            self._minio.list_buckets()
+            return True
+        except BaseException:
+            return False
+
+    is_accessible.__doc__ = Storage.is_accessible.__doc__
+
+    def pull_job_data(self, job_id: str, target_directory: str) -> None:
+        pull_minio_bucket(self._minio, job_id, target_directory)
```

Comment:
We should wrap the `MinioError`s with our `StorageError`s (in all the `MinioStorage` functions).

Reply:
I just found out they are actually wrapped by `pull_minio_bucket` and friends.